### PR TITLE
Add more tests and fix capabilities bug

### PR DESCRIPTION
**Pull request by rrooij**
Some were already covered and others have been added to make some routes.pl tests unnecessary.

I also fixed a bug with regards to authentication. It would fail certain endpoints when tested with header authentication.

### CODE REVIEW

**Comment by rrooij** [comment]
Wow, a lot more autoloading issues came up since I started working on this. I will rebase from main and try to resolve the remaining ones. :smile:

**Comment by spl** [comment]
Am I correct in understanding that this is pending on #962?

**Comment by rrooij** [comment]
I think it is ready for review now. I want to convert more tests but I do think that it is worth merging this already in advance because there are important bug fixes like the `capabilities_bug` as well.

**Comment by spl** [comment]
> I've got no clue how these other commits ended up here? I rebased the branch from main and it apparently caused some problems.

I don't know either. I added a commit, rebased to `main`, and force-pushed. Now the changes to `src/core/document/diff.pl` don't show up.

**Comment by spl** [comment]
I'll look at this in more detail tomorrow.

File: tests/test/triples.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    @prefix owl: <http://www.w3.org/2002/07/owl#> .
+    layer:LayerIdRestriction a owl:Restriction.`
+    const secondReq = await agent.put(path).send({
+      commit_info: { author: 'TestSuite', message: 'Testing Turtle triples 2' },
+      turtle: secondTurtle,
+    })
+    expect(secondReq.status).to.equal(200)
+  })
+
+  // TODO: Create test for this, it responds with an application/json type now... which it isn't.
```
Comment (spl) [inline tests/test/triples.js:61]: Please create issues for these TODOs and reference them in the comments here.

File: tests/lib/agent.js [hunk 2]
```diff
@@ -29,16 +29,20 @@ class Agent {
   // Add authentication
   auth () {
     this.userName = process.env.TERMINUSDB_USER
-    assert(this.userName, 'Missing environment variable: TERMINUSDB_USER')
```
Comment (spl) [inline tests/lib/agent.js:32]: Why do you remove this?

File: tests/lib/agent.js [hunk 3]
```diff
@@ -29,16 +29,20 @@ class Agent {
...
         request.auth(token, { type: 'bearer' })
       })
+    } else if (insecureUserHeader) {
+      this.agent.use((request) => {
+        request.set(insecureUserHeader, this.userName)
+      })
     } else {
       const pass = process.env.TERMINUSDB_PASS
-      assert(pass, 'Missing environment variable: TERMINUSDB_ACCESS_TOKEN or TERMINUSDB_PASS')
+      assert(pass, 'Missing environment variable: TERMINUSDB_ACCESS_TOKEN, TERMINUSDB_INSECRE_USER_HEADER or TERMINUSDB_PASS')
```
Comment (spl) [inline tests/lib/agent.js:45]:
```suggestion
      assert(pass, 'Missing environment variable: TERMINUSDB_ACCESS_TOKEN, TERMINUSDB_INSECURE_USER_HEADER, or TERMINUSDB_PASS')
```

**Review by spl — CHANGES_REQUESTED** [review]
I'm still reviewing. Here a few comments.

File: tests/lib/agent.js (hunk 2 above)
Comment (rrooij) [reply tests/lib/agent.js:32]: Because the JWT auth does not need a user, since the user is already encoded in the token.

Comment (spl) [reply tests/lib/agent.js:32]: But that's only for JWT. `this.userName` is used twice below. What happens when it's `undefined` in those cases?

```js
      this.agent.use((request) => {
        request.set(insecureUserHeader, this.userName)
      })
```

```js
      this.agent.use((request) => {
        request.auth(this.userName, pass)
      })
```

Comment (rrooij) [reply tests/lib/agent.js:32]: You're right. It should conditionally assert it instead. :smile:

Comment (rrooij) [reply tests/lib/agent.js:32]: I think I fixed it in the latest commit.

**Review by spl — APPROVED** [review]

**Pull request merged**